File: lista3/java/Main.java
public class Main {
    public static void main(String[] args) {
        DHSetup dhSetup = new DHSetup();

        User alice = new User(dhSetup);
        User bob = new User(dhSetup);

        System.out.println("prime: " + GF.characteristic());
        System.out.println("generator: " + dhSetup.getGenerator());

        System.out.println("Alice's public key: " + alice.getPublicKey());
        System.out.println("Bob's public key: " + bob.getPublicKey());

        alice.setKey(bob.getPublicKey());
        bob.setKey(alice.getPublicKey());

        GF message = new GF(1234); // Wiadomość do zaszyfrowania

        GF encrypted = alice.encrypt(message);
        System.out.println("Encrypted message: " + encrypted);

        GF decrypted = bob.decrypt(encrypted);
        System.out.println("Decrypted message: " + decrypted);
    }
}

File: lista3/python/main.py
# Main.py

from DHSetup import DHSetup
from User import User
from GF import GF

if __name__ == "__main__":
    dh_setup = DHSetup()

    alice = User(dh_setup)
    bob = User(dh_setup)

    print("prime:", dh_setup.prime)
    print("generator:", dh_setup.get_generator())

    print("Alice's public key:", alice.get_public_key())
    print("Bob's public key:", bob.get_public_key())

    alice.set_key(bob.get_public_key())
    bob.set_key(alice.get_public_key())

    message = GF(12345)

    encrypted = alice.encrypt(message)
    print("Encrypted message:", encrypted)

    decrypted = bob.decrypt(encrypted)
    print("Decrypted message:", decrypted)
